async.c

```diff
@@ -782,6 +809,7 @@ static int __redisAsyncCommand(redisAsyncContext *ac, redisCallbackFn *fn, void
             if (de != NULL) {
                 existcb = dictGetEntryVal(de);
                 cb.pending_subs = existcb->pending_subs + 1;
+                __redisRunFinalizer(ac,existcb);
```

Comment:
Wouldn't it be prudent to check here if the contents of the callback is different from the replacement?  In particular the `privdata` member, but probably also the callback and finalizer, in case the caller uses dynamically created stubs and `NULL` for `privdata`.